### ISSUE TITLE
bug: Bunny infinetely loading after first startup

### ISSUE THREAD

**Issue by 404unknownCoder** [pyoncord/Bunny#72]
**Describe the bug**
Bunny infinitely loads after closing the app before

**To Reproduce**
1. Open bunny
2. Close the app 
3. Open it again
4. Infinitely showing the logo

**Expected behavior**
Show my dm list (start up)

**Screenshots**
![image](https://github.com/user-attachments/assets/e80a747a-2689-454c-8f4b-397645482de0)

**Device Information (please complete the following information):**
 - Device: [Poco X3 Pro (M2102J20SG)]
 - OS: [Crdroid 10 (Android 14)] -> stock MIUI worked fine though
 - Discord Version [24214] (tested on stable too)
 - Bunny Version [e9d80c7-main] (tested on stable too)

**Checklists**
- [x] I understand that this is a bug report and not a place to request new features or plugins
- [x] I understand that this is no place to report bugs produced by a plugin
- [x] I can reproduce this bug in Bunny's safe mode AND unable to reproduce it in vanilla/unmodded Discord

**Comment by 404unknownCoder** [comment]
i have tried reflashing the rom, using lineageos which works fine (other people use it without issues) reinstalling, and doing basic things like resetting using different internet connection, ...

also seems to be unreproducable unless you have a poco x3 pro too, as someone else using lineageos can use bunny without issues so incase logs or something is needed i have no issues with providing info/helping with finding the issue

**Comment by SerpentDash** [comment]
Same issue here on A14 Crdroid 10.6 - Oneplus 7T Pro

As previously mentioned, the first launch works fine, but every subsequent attempt to open the app causes it to freeze on the splash screen. After rebooting, the app is again able to launch, but only once.
For the record, Bunny works fine on an older version of this custom ROM (10.2). Every newer version causes the issue described above.

Logcat:
> 08-06 13:52:28.667  1667  3733 I ActivityTaskManager: START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=io.github.pyoncord.app/com.discord.main.MainDefault bnds=[576,733][864,1105]} with LAUNCH_SINGLE_TASK from uid 10442 (BAL_ALLOW_ALLOWLISTED_COMPONENT) result code=0
08-06 13:52:28.667  1667  1753 W ActivityManager: ProcessRecord{28a6c7 0:io.github.pyoncord.app/u0a487} is attached to a previous process 14308
08-06 13:52:28.668  3208  3252 V WindowManagerShell: Transition requested (#250): android.os.BinderProxy@2f323a9 TransitionRequestInfo { type = OPEN, triggerTask = TaskInfo{userId=0 taskId=38 displayId=0 isRunning=true baseIntent=Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=io.github.pyoncord.app/com.discord.main.MainDefault } baseActivity=ComponentInfo{io.github.pyoncord.app/com.discord.main.MainActivity} topActivity=ComponentInfo{io.github.pyoncord.app/com.discord.main.MainActivity} origActivity=ComponentInfo{io.github.pyoncord.app/com.discord.main.MainDefault} realActivity=ComponentInfo{io.github.pyoncord.app/com.discord.main.MainActivity} numActivities=1 lastActiveTime=4622145 supportsMultiWindow=true resizeMode=1 isResizeable=true minWidth=-1 minHeight=-1 defaultMinSize=220 token=WCT{android.window.IWindowContainerToken$Stub$Proxy@419dd2e} topActivityType=1 pictureInPictureParams=null shouldDockBigOverlays=false launchIntoPipHostTaskId=-1 lastParentTaskIdBeforePip=-1 displayCutoutSafeInsets=null topActivityInfo=ActivityInfo{ffb60cf com.discord.main.MainDefault} launchCookies=[android.os.BinderProxy@4a17a5c] positionInParent=Point(0, 0) parentTaskId=-1 isFocused=false isVisible=false isVisibleRequested=false isSleeping=false locusId=null displayAreaFeatureId=1 isTopActivityTransparent=false appCompatTaskInfo=AppCompatTaskInfo { topActivityInSizeCompat=false topActivityEligibleForLetterboxEducation= false isLetterboxDoubleTapEnabled= false topActivityEligibleForUserAspectRatioButton= false topActivityBoundsLetterboxed= false isFromLetterboxDoubleTap= false topActivityLetterboxVerticalPosition= -1 topActivityLetterboxHorizontalPosition= -1 topActivityLetterboxWidth=1440 topActivityLetterboxHeight=3120 isUserFullscreenOverrideEnabled=false isSystemFullscreenOverrideEnabled=false cameraCompatControlState=hidden}}, pipTask = null, remoteTransition = RemoteTransition { remoteTransition = android.window.IRemoteTransition$Stub$Proxy@b315165, appThread = android.app.IApplicationThread$Stub$Proxy@3ea9e3a, debugName = QuickstepLaunch }, displayChange = null, flags = 0, debugId = 250 }
08-06 13:52:28.673  1667  1753 W ActivityManager: ProcessRecord{28a6c7 0:io.github.pyoncord.app/u0a487} is attached to a previous process 14308
08-06 13:52:28.674  1667  1753 W ActivityManager: ProcessRecord{28a6c7 0:io.github.pyoncord.app/u0a487} is attached to a previous process 14308
08-06 13:52:28.684  1667  5468 D CoreBackPreview: Window{c4b84eb u0 Splash Screen io.github.pyoncord.app}: Setting back callback OnBackInvokedCallbackInfo{mCallback=android.window.IOnBackInvokedCallback$Stub$Proxy@4c4b606, mPriority=0, mIsAnimationCallback=false}
08-06 13:52:28.692  1667  1748 V WindowManager: Sent Transition (#250) createdAt=08-06 13:52:28.657 via request=TransitionRequestInfo { type = OPEN, triggerTask = TaskInfo{userId=0 taskId=38 displayId=0 isRunning=true baseIntent=Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=io.github.pyoncord.app/com.discord.main.MainDefault } baseActivity=ComponentInfo{io.github.pyoncord.app/com.discord.main.MainActivity} topActivity=ComponentInfo{io.github.pyoncord.app/com.discord.main.MainActivity} origActivity=ComponentInfo{io.github.pyoncord.app/com.discord.main.MainDefault} realActivity=ComponentInfo{io.github.pyoncord.app/com.discord.main.MainActivity} numActivities=1 lastActiveTime=4622145 supportsMultiWindow=true resizeMode=1 isResizeable=true minWidth=-1 minHeight=-1 defaultMinSize=220 token=WCT{RemoteToken{e783260 Task{80b1d30 #38 type=standard A=10487:io.github.pyoncord.app}}} topActivityType=1 pictureInPictureParams=null shouldDockBigOverlays=false launchIntoPipHostTaskId=-1 lastParentTaskIdBeforePip=-1 displayCutoutSafeInsets=null topActivityInfo=ActivityInfo{4553a73 com.discord.main.MainDefault} launchCookies=[android.os.BinderProxy@ec6e319] positionInParent=Point(0, 0) parentTaskId=-1 isFocused=false isVisible=false isVisibleRequested=false isSleeping=false locusId=null displayAreaFeatureId=1 isTopActivityTransparent=false appCompatTaskInfo=AppCompatTaskInfo { topActivityInSizeCompat=false topActivityEligibleForLetterboxEducation= false isLetterboxDoubleTapEnabled= false topActivityEligibleForUserAspectRatioButton= false topActivityBoundsLetterboxed= false isFromLetterboxDoubleTap= false topActivityLetterboxVerticalPosition= -1 topActivityLetterboxHorizontalPosition= -1 topActivityLetterboxWidth=1440 topActivityLetterboxHeight=3120 isUserFullscreenOverrideEnabled=false isSystemFullscreenOverrideEnabled=false cameraCompatControlState=hidden}}, pipTask = null, remoteTransition = RemoteTransition { remoteTransition = android.window.IRemoteTransition$Stub$Proxy@2367fde, appThread = android.app.IApplicationThread$Stub$Proxy@fde45bf, debugName = QuickstepLaunch }, displayChange = null, flags = 0, debugId = 250 }
08-06 13:52:28.692  1667  1748 V WindowManager:         {WCT{RemoteToken{e783260 Task{80b1d30 #38 type=standard A=10487:io.github.pyoncord.app}}} m=OPEN f=NONE leash=Surface(name=Task=38)/@0x5e62c92 sb=Rect(0, 0 - 1440, 3120) eb=Rect(0, 0 - 1440, 3120) d=0},
08-06 13:52:38.677  1667  1778 E ActivityManager: ProcessRecord{28a6c7 0:io.github.pyoncord.app/u0a487} 14308 refused to die while trying to launch ProcessRecord{3b60b65 0:io.github.pyoncord.app/u0a487}, cancelling the process start
08-06 13:53:21.105  1667  2279 D ActivityManager: sync unfroze 16319 io.github.pyoncord.manager for 6
08-06 13:53:46.750  1667  2462 W InputDispatcher: Not sending touch gesture to ff913ad ActivityRecordInputSink io.github.pyoncord.app/com.discord.main.MainActivity because it has config NO_INPUT_CHANNEL

**Comment by SerpentDash** [comment]
Some other findings:

When the app is in 'frozen state' on Splash Screen, `adb shell "ps | grep io.github.pyoncord.app"` command shows two processes of this app:

> u0_a511       9163     1   32105212 614512 0                   0 S io.github.pyoncord.app
> u0_a511       9164     1   32105212 614512 0                   0 S io.github.pyoncord.app

Closing the app (using GUI or adb) won't do much. **However**, after using `adb shell "pkill -9 -f io.github.pyoncord.app"` app will start launching (or will launch after opening app again). Unfortunately this command **requires root access**...

**Comment by dazziedez** [comment]
Other roms based on QPR2 seem to be facing the same behavior afaik, so it might be an issue with aosp since [crdroid rebased to QPR2 on v10.3](https://crdroid.net/blog/2024-04-05-crDroid-10.3-released).

I'm not smart by and means, but the logs provided show that:
1. there are multiple warnings about the process being attached to the secret, more sinister process:
```
08-06 13:52:28.667 1667 1753 W ActivityManager: ProcessRecord{28a6c70:io.github.pyoncord.app/u0a487} is attached to a previous process 14308
```

2. the evil process in question is not properly stopping when it should, preventing a clean restart (Which is resolved on reboot):
```
08-06 13:52:38.677 1667 1778 E ActivityManager: ProcessRecord{28a6c70:io.github.pyoncord.app/ua487} 14308 refused to die while trying to launch ProcessRecord{3b60b650:io.github.pyoncord.app/u0a487}, cancelling the process start
```

I hope that helps anyone!!! Sorry if I'm dumb and it's wrong...

**Comment by Ermageeerd** [comment]
> Closing the app (using GUI or adb) won't do much. **However**, after using `adb shell "pkill -9 -f io.github.pyoncord.app"` app will start launching (or will launch after opening app again). **Unfortunately** this command **requires root access**...

Shizuku allows the nonroot usage of ADB.

**Comment by DaRandomCube** [comment]
> > Closing the app (using GUI or adb) won't do much. **However**, after using `adb shell "pkill -9 -f io.github.pyoncord.app"` app will start launching (or will launch after opening app again). **Unfortunately** this command **requires root access**...
> 
> Shizuku allows the nonroot usage of ADB.

Tried using the (use shizuku in terminal apps)
After you configure and run `rish` you will be dropped into a shell, trying to run the command won't work sadly

**Comment by Ermageeerd** [comment]
> Tried using the (use shizuku in terminal apps) After you configure and run `rish` you will be dropped into a shell, trying to run the command won't work sadly

Did you get the package name or the displayed name?
(example: Termux is com.termux)

**Comment by DaRandomCube** [comment]
> > Tried using the (use shizuku in terminal apps) After you configure and run `rish` you will be dropped into a shell, trying to run the command won't work sadly
> 
> Did you get the package name or the displayed name? (example: Termux is com.termux)

yea
decided to root anyway

**Comment by Ermageeerd** [comment]
> yea decided to root anyway

ok, your decision, be careful and have fun! :)

**Comment by amsryq** [comment]
closing as this is more to a rom/external issue
